Extensions/TileMap/JsExtension.js

```diff
@@ -479,8 +477,8 @@ module.exports = {
         this._pixiObject.dirty = true;
       }
 
-      this._pixiObject.x = this._instance.getX() + this._pixiObject.width / 2;
-      this._pixiObject.y = this._instance.getY() + this._pixiObject.height / 2;
+      this._pixiObject.x = this._instance.getX();
+      this._pixiObject.y = this._instance.getY();
```

Comment:
Double checking this, are you sure it's what you want to do? Your changes implies that the object will rotate from the top-left instead of the center.

I'm not even sure a tilemap can be rotated, but still I'm not sure if these are the correct changes? :) 
Let me know!

Reply:
this is offsetting it from its origin

![Capture](https://user-images.githubusercontent.com/6495061/91657264-9355a600-eab7-11ea-9e8d-4b60b7c4b836.PNG)


but you are right, now it rotates from the top left corner. Should I do something else?

Reply:
In other objects, we set anchor.x/anchor.y. But this object lacks support for anchor.
Either we get `anchor` added to Pixi.tilemap, or we do the calculation by ourselves by setting:

```
this._pixiObject.pivot.x = this._pixiObject.width / 2;
this._pixiObject.pivot.y = this._pixiObject.height / 2;
```

(and putting back:
```
this._pixiObject.x = this._instance.getX() + this._pixiObject.width / 2;
this._pixiObject.y = this._instance.getY() + this._pixiObject.height / 2;
```
)

but this seems not to work in the case of a resized object, surely because the width/height are changed (there is an offset when you resize).

Reply:
I think we need to find a way to get the size of the object *before* any resize is made, i.e: find the original width and original height.

Reply:
Ok I think I found the solution :) Let me push it.

Reply:
Thankyou @4ian  :) feel free to push it

Reply:
I've pushed the changes :)

The idea is:

- We want the (X,Y) coordinates of the object to be the top-left. Why? Because it's the default in GDevelop and because the alternative is to put the (X,Y) coordinates to be the center of the object, which make sense for particles/lights but not for a tilemap. So far, no problems, it's the default in Pixi Tilemap and in GDevelop instances.
- We want the rotation to be done from the center of the object. Why? Because it's the default in GDevelop (apart for sprites where you can customize this) and it's easier for users and 99% of use cases. For this, we can:
  - usually, just set `anchor` x and y to 0.5;0.5. Unfortunately, this does not exist for PIXI tilemap.
  - instead, we have to use `pivot` to set the center (which is doable for all Pixi objects). CAREFUL though, this `pivot` is using pixels! i.e: you must enter something like (200,150) if your object size is (400,300). For this, we need to get the width and height of the object, then divide by 2. CAREFUL though, we need the original size of the object. We can't use directly `this._pixiObject.width`, because the width is scaled. Thankfully, it's pretty simple, we just divide by the scale (internally, when you set the width of an object in Pixi, Pixi just changes the scale): `this._pixiObject.width / this._pixiObject.scale.x`.
- Cool, we now have the rotation made in the center of the object! Last stuff, by changing the `pivot` (or the `anchor`), we also changed for Pixi the origin of the object. So the (X,Y) position in GDevelop of the instance is now corresponding to the center! The solution is easy, just convert from the top-left coordinates to the center coordinates by adding half the size (using `this._pixiObject.width` because we add half the size AFTER scaling).

Lots of stuff to get right, but seems it works well now :)

The rendered pixi object is properly following the size of the instance: 
![image](https://user-images.githubusercontent.com/1280130/91658841-a9696380-eac3-11ea-94a1-d50a303eabf6.png)

and the rotation is done around the center:
![image](https://user-images.githubusercontent.com/1280130/91658848-be45f700-eac3-11ea-8680-382761f28b6c.png)

(for the blue selection rectangle, remember that currently in GDevelop editor it is not rotation ^^' But that's a GDevelop improvement to do, the object is rotating properly :))

Reply:
@blurymind: last weird stuff, when I click on the object, it's not getting selected. I wonder if it's missing some interaction/click handler somewhere.
For now I have to select it using the list of instances.

Apart from this, the resize and rotation works well though, which is the most important stuff! :)

Reply:
tilemap does not have containsPoint() implementation, because, obviously, it will check **all the tiles every frame**. That's why i didnt add it in main repo.

Reply:
I also advice to rotate stuff around left-top corner and use `pivot` if you need to move that point. `tilemap.pivot = supposed_center_in_local_coords` , you can calc center using width/height or whatever. 

Every time people use `width`/`height` I post the same advice: do read the source code of pixi Container, Sprite, to see how width/height are handled. Those are second-class citizens field, they are calculated. I think people rely on pixi magic too much, there's no magic, width/height are breaking very often. Without understanding of `getBounds()` , `getLocalBounds()` you can make many mistakes.

Reply:
Also, tilemap rotating around (0,0) by default because everyone uses (0,0) as left-top corner. I dont know why people dont use negative coords for tiles.

Reply:
I see! We rely on the `click` event in GDevelop, but that's no big deal for now. 
Either we can add a click handler to Pixi Tilemap by adding containsPoint that just checks if the click is made inside the (potentially rotated) bounding rectangle of the object or we'll switch our implementation to use the position + size + rotation of our GDevelop instances (i.e: the equivalent of the blue rectangle in my screenshots) - so that we're independent from Pixi objects being able to be clicked or not.

@blurymind I advise to concentrate on getting the Runtime version properly and cleanly done as a priority - what I explained here is potentially a refactoring that I'll do in GDevelop (unless click is implemented for Pixi tilemap in the meantime).

Reply:
Sorry, missed your additional messages, GitHub did not show them!

> tilemap.pivot = supposed_center_in_local_coords , you can calc center using width/height or whatever.

Yep that's what we end up doing, thanks for confirming! :) 

I think we're good, there was a bit of confusion around width/height because as you said there are derived fields from the scale. I'll think about using `this.getLocalBounds().width` and `this.getLocalBounds().height` directly, much better! :) 
Thanks for the tips!

Reply:
yeah, directly is more understandable, and also its *before scale* , real local width